cmd/api.go

```diff
@@ -163,41 +163,15 @@ func (o *apiOptions) validate(c *config.Config) error {
 
 	// In case we want to scaffold a resource API we need to do some checks
 	if o.doResource {
-		// Skip the following check for v1 as resources aren't tracked
-		if !c.IsV1() {
-			// Check that resource doesn't exist or flag force was set
-			if !o.force {
-				resourceExists := false
-				for _, r := range c.Resources {
-					if r.Group == o.resource.Group &&
-						r.Version == o.resource.Version &&
-						r.Kind == o.resource.Kind {
-						resourceExists = true
-						break
-					}
-				}
-				if resourceExists {
-					return errors.New("API resource already exists")
-				}
-			}
+		// Check that resource doesn't exist or flag force was set
+		if !o.force && c.HasResource(o.resource.GVK()) {
+			return errors.New("API resource already exists")
 		}
 
-		// The following check is v2 specific as multi-group isn't enabled by default
-		if c.IsV2() {
-			// Check the group is the same for single-group projects
-			if !c.MultiGroup {
-				validGroup := true
-				for _, existingGroup := range c.ResourceGroups() {
-					if !strings.EqualFold(o.resource.Group, existingGroup) {
-						validGroup = false
-						break
-					}
-				}
-				if !validGroup {
-					return fmt.Errorf("multiple groups are not allowed by default, to enable multi-group visit %s",
-						"kubebuilder.io/migration/multi-group.html")
-				}
-			}
+		// Check that the provided group can be added to the project
+		if c.IsV2() && !c.MultiGroup && len(c.Resources) != 0 && !c.HasGroup(o.resource.Group) {
```

Comment:
```suggestion
		if c.IsV2() && !c.MultiGroup && !c.HasGroup(o.resource.Group) {
```

Reply:
`c.IsV2()` is needed. Imagine you do a `kubebuilder init --version=1 ...` and then `kubebuilder create api ...`. `!c.MultiGroup && !c.HasGroup(o.resource.Group)` will be true and print the error when it shouldn't.

`len(c.Resources) != 0` is needed for the first resource. Imagine you do a `kubebuilder init ...` then  a `kubebuilder create api ...`. `!c.MultiGroup && !c.HasGroup(o.resource.Group)` will be true printing the error when it is the first resource so it should be admited.